core/src/main/scala/sttp/tapir/internal/package.scala

```diff
@@ -192,6 +191,22 @@ package object internal {
         Vector(b.bodyType)
       }.headOption
     }
+
+    def matchesContent(content: MediaType): Boolean = {
+      val contentWithCharset = content match {
+        case m @ MediaType(_, _, None) => m.charset(StandardCharsets.UTF_8.name()) // default UTF-8
```

Comment:
I don't think we should add UTF-8 as the default. Quite the contrary, most media types *except* `text/*` don't support charsets

Reply:
right, only text has a default one